omise/classes/charge.php

```diff
@@ -0,0 +1,64 @@
+<?php
+if (! defined('_PS_VERSION_')) {
+    exit();
+}
+
+if (defined('_PS_MODULE_DIR_')) {
+    require_once _PS_MODULE_DIR_ . '/omise/libraries/omise-php/lib/Omise.php';
+    require_once _PS_MODULE_DIR_ . '/omise/libraries/omise-plugin/Omise.php';
+    require_once _PS_MODULE_DIR_ . '/omise/setting.php';
+}
+
+class Charge extends ModuleFrontController
+{
+    public function create()
+    {
+        $charge_request = array(
+            'amount' => $this->getAmount(),
+            'card' => $this->getCardToken(),
+            'capture' => $this->getCapture(),
+            'currency' => $this->getCurrencyCode(),
+            'description' => $this->getChargeDescription(),
+        );
+
+        return OmiseCharge::create($charge_request, '', $this->getSecretKey());
+    }
+
+    protected function getAmount()
+    {
+        $currency_code = $this->getCurrencyCode();
+        $order_total = (float) $this->context->cart->getOrderTotal(true, Cart::BOTH);
+
+        return OmisePluginHelperCharge::amount($currency_code, $order_total);
+    }
+
+    protected function getCardToken()
+    {
+        return Tools::getValue('omise_card_token');
+    }
+
+    protected function getCapture()
```

Comment:
@guzzilar The payment method will be developed along with the development of a feature, manual capture.